File: plugin/testDataOutput2/shared/lambda.swift
//Package: com.test
//Converted using Kwift2

import Foundation


public func lambdaTest() -> Void {
    takesLambda(lambda: { (integer, someValue) in 
        print("Hello!")
    }
)
    takesLambda{ (integer, someValue) in 
        print("Hello!")
    }
}
 
 
public func takesLambda(lambda: (integer: Int, someValue: String) -> Void) -> Void {
}
 
 
public func takesEscapingLambda(lambda: @escaping (integer: Int, someValue: String) -> Void) -> Void {
}
 
 
public func takesEscapingLambda2(lambda: (@escaping (integer: Int, someValue: String) -> Void)) -> Void {
}
 
 
public class ConstructorHasEscaping {
    
    public var lambda:  (integer: Int, someValue: String) -> Void
    
    init(lambda: @escaping (integer: Int, someValue: String) -> Void) {
        self.lambda = lambda
    }
    
}
 
 
 
public class ConstructorHasEscaping2 {
    
    public var lambda: ( (integer: Int, someValue: String) -> Void)
    
    init(lambda: (@escaping (integer: Int, someValue: String) -> Void)) {
        self.lambda = lambda
    }
    
}
 

File: plugin/testData/storage/lambda.kt
package com.test

fun lambdaTest() {
    takesLambda(lambda = { integer, someValue ->
        println("Hello!")
    })
    takesLambda { integer, someValue ->
        println("Hello!")
    }
}

fun takesLambda(lambda: (integer: Int, someValue: String) -> Unit) {

}

fun takesEscapingLambda(@escaping lambda: (integer: Int, someValue: String) -> Unit) {

}

class ConstructorHasEscaping(@escaping val lambda: (integer: Int, someValue: String) -> Unit) {
}
